internal/uvm/delete_container.go

```diff
@@ -0,0 +1,13 @@
+package uvm
+
+import (
+	"context"
+)
+
+func (uvm *UtilityVM) DeleteContainerState(ctx context.Context, cid string) error {
+	if uvm.gc == nil || !uvm.guestCaps.DeleteContainerStateSupported {
```

Comment:
I would prefer we check DeleteContainerSupported in the caller so that its easier to see. We can leave this code here and return an error if its not supported as a safety check but It really seems clearer in the caller to write:

```go
if vm.SupportsDeleteContainer() {
    vm.DeleteContainer(ctx, id)
}
```

What do ya think?

Reply:
I think it may be a bit redundant, but I'm okay with it if you think it would be clearer.

Reply:
I just feel its kindof an anti-pattern to read code that says:
```go
if err := DoThis(); err != nil {
    return err
}
if err := DoThat(); err != nil {
    return err
}
if err := DoMore(); err != nil {
    return err
}
```
And expect that all three funcs might do nothing and also return no error because `This`, `That`, and `More` are not supported. Seems like its much more clear to say from the caller
```go
if DoThisSupported() {
    if err := DoThis(); err != nil {
        return err
    }
}
```
Or also this is a common pattern that removes the need of the caller to know but also as the func return an error when its not supported so the caller can ignore.
```go
if err := DoThis(); err != nil && !isNotSupportedError(err) {
    return err
}
```